tools/gen/impl_template

```diff
@@ -305,6 +305,39 @@ fail:
 	fromwire_fail(cursor, max);
 	return false;
 }
+bool ${tlv.name}_is_valid(const struct ${tlv.struct_name()} *record)
+{
+	size_t numfields = tal_count(record->fields);
+	bool first = true;
+	u64 prev_type = 0;
+	for (size_t i=0; i<numfields; i++) {
+		struct tlv_field *f = &record->fields[i];
+		if (f->numtype % 2 == 0 && f->meta == 0) {
```

Comment:
f->meta == NULL please.

Reply:
Ouch, sorry. Copy pasted it without checking.